src/libcore/iter/mod.rs

```diff
@@ -1643,20 +1708,145 @@ impl<I: DoubleEndedIterator, U, F> DoubleEndedIterator for FlatMap<I, U, F> wher
 ///
 /// [`fuse()`]: trait.Iterator.html#method.fuse
 /// [`Iterator`]: trait.Iterator.html
-#[derive(Clone, Debug)]
 #[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct Fuse<I> {
-    iter: I,
-    done: bool
+    iter: <I as SpecFuse>::T,
```

Comment:
Oh dear (https://github.com/rust-lang/rust/issues/35727).

Well, I *thought* that I could do the same thing as the `Zip` specialization and put the flag in a separate static data field (keeping the iterator in the main struct) but apparently that doesn't appease rust:

```rust
struct Fuse<I> {
    iter: I,
    fused: <I as FuseImpl>::Data,
}
```

There must be some way to do this...